pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -479,6 +503,22 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 
 		// fill in the collect stats IP address of the App
 		appInstance.CollectStatsIPAddr = net.ParseIP(cfgApp.GetCollectStatsIPAddr())
+		ipAddr := cfgApp.GetLocalDataStoreIPAddr()
+		if ipAddr != "" {
+			appInstance.LocalDataStoreIPAddr = net.ParseIP(ipAddr)
+			for _, ds := range dsItems {
+				configds := ds.(types.DatastoreConfig)
+				if configds.IsLocal {
+					continue
+				}
+				if isHostInFqdn(configds.Fqdn, ipAddr) {
+					log.Functionf("parseAppInstanceConfig: %s, add local flag", configds.Fqdn)
+					configds.IsLocal = true
+					getconfigCtx.pubDatastoreConfig.Publish(configds.Key(), configds)
+					break
+				}
+			}
```

Comment:
Might make sense to definte some function(s) to check and set IsLocal; updateDataStoreIsLocal() type thing - here and for the clearing of LocalDataStoreIPAddr. Maybe it can walk all app instances and all datastores and set/clear IsLocal thus one function for set and clear? Or two separate functions would also be ok.

Reply:
Will do.